plugins/reach/lib/model/EntryVendorTask.php

```diff
@@ -235,9 +240,84 @@ public function getServiceFeature()
 		return $this->getFromCustomData(self::CUSTOM_DATA_SERVICE_FEATURE);
 	}
 
-	/* (non-PHPdoc)
- 	 * @see BaseEntryVendorTask::preSave()
- 	 */
+	protected function calculateNextBusinessDay($currentDay, $currentTime)
+	{
+
+		$endDayString = sprintf("Y-m-d %s:00:00", self::BUSINESS_DAY_END_HOUR);
+
+		if( $currentDay > self::BUSINESS_DAY_FRIDAY )
+		{
+			//Weekend
+			$daysToSkip = self::BUSINESS_DAY_NEXT_MONDAY - $currentDay;
+			$expTime = date($endDayString, strtotime("+$daysToSkip days"));
+		}
+		else
+		{
+			$endDayTime = self::BUSINESS_DAY_END_HOUR * self::BUSINESS_DAY_TIME_NORMALIZATION_FACTOR;
+			$startDayTime = self::BUSINESS_DAY_START_HOUR * self::BUSINESS_DAY_TIME_NORMALIZATION_FACTOR;
+
+			//Middle of a business day - expiration time: add 24 hours
+			if( ($startDayTime <= $currentTime) && ($currentTime <= $endDayTime) )
+			{
+				$expTime = date("Y-m-d H:i:s", strtotime('+1 days'));
+			}
+			//Before work hours - expiration time: End Day today
+			elseif ($currentTime < $startDayTime)
+			{
+				$expTime = date($endDayString, strtotime('now'));
+			}
+			//After work hours - expiration time: End Day tomorrow
+			else
+			{
+				$expTime = date($endDayString, strtotime('+1 days'));
+			}
+		}
+
+		return $expTime;
+	}
+
+	protected function calculateBusinessDays($numBusinessDays)
+	{
+		//Get Local TZ
+		$localTimeZone = date_default_timezone_get();
+
+		//Set to EST Time
+		date_default_timezone_set('America/New_York');
+
+		$currentDateTime = new DateTime("now");
+		$currentDay = date("N");
+		$currentTime = date("His");
+
+		//Calculate the next (1st) business day
+		$expTime = $this->calculateNextBusinessDay($currentDay, $currentTime);
+
+		//Calculate the other business days
+		if ($numBusinessDays > 1)
```

Comment:
Why not just do - 

while(numBusinessDays)
{
       $time = $this->calculateNextBusinessDay($day, $cme);
      //calculate the day
     numBusinessDays--
}

In the current code you are limited up to 5 business days.